models/issues/pull.go

```diff
@@ -504,13 +504,27 @@ func NewPullRequest(outerCtx context.Context, repo *repo_model.Repository, issue
 	defer committer.Close()
 	ctx.WithContext(outerCtx)
 
+	// onFail will try to gracefully revert the update in the resource index.
+	onFail := func() error {
+		// Close commiter.
+		committer.Close()
+		// Try to revert the increase in resource index.
+		if err := db.UpsertDecrResourceIndex(outerCtx, "issue_index", repo.ID, idx); err != nil {
```

Comment:
I suspect that outerCtx isn't the correct thing here. In fact I'm certain it isn't.

`db.DefaultContext` is needed here.